src/ObjectUtils/include/ObjectUtils/PdbDiaUtils.h

```diff
@@ -0,0 +1,352 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef OBJECT_UTILS_PDB_DIA_UTILS_H_
+#define OBJECT_UTILS_PDB_DIA_UTILS_H_
+
+#include <Windows.h>
+#include <atlbase.h>
+#include <cguid.h>
+#include <dia2.h>
+
+#include <string>
+
+#include "OrbitBase/Logging.h"
+
+// Provides utility functions to retrieve a functions parameter types as string.
+namespace orbit_object_utils {
+
+// Retrieves a string representing the parameter list of the given function or
+// function type. For functions with function type "<no type>" this returns an
+// empty string.
+std::string PdbDiaParameterListAsString(IDiaSymbol* function);
+
+// Retrieves a string representation of the give type symbol. It will prepend the
+// given `pointer_type_string` to the result string, or in case of a function pointer,
+// add it inside the paranthesis (e.g. void(*)(int, int)).
+// If the type can't be computed properly, "???" will be returned.
+std::string PdbDiaTypeAsString(IDiaSymbol* type, std::string pointer_type_str = "");
+
+std::string PdbDiaParameterListAsString(IDiaSymbol* function_or_function_type) {
+  HRESULT result;
+
+  DWORD tag;
+  ORBIT_CHECK(SUCCEEDED(function_or_function_type->get_symTag(&tag)));
```

Comment:
Have you considered making the function return an `ErrorMessageOr<std::string>` and returning an error instead of aborting Orbit with ORBIT_CHECKs? The error message from the Dia SDK could be returned through `GetLastErrorAsString()` and it seems we'd have a better idea of what went wrong. We could always assert one level higher, but we'd log the actual error. We could also not assert, but simply log the error, wdyt?